### PR TITLE
feat: add support for duckdb style "FROM <tbl>" statements

### PR DESCRIPTION
**Pull request by universalmind303**
allows for parsing of duckdb's `from` queries

closes https://github.com/sqlparser-rs/sqlparser-rs/issues/1143

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -6722,6 +6726,8 @@ impl<'a> Parser<'a> {
             SetExpr::Values(self.parse_values(is_mysql)?)
         } else if self.parse_keyword(Keyword::TABLE) {
             SetExpr::Table(Box::new(self.parse_as_table()?))
+        } else if self.parse_keyword(Keyword::FROM) {
```
Comment (universalmind303) [inline src/parser/mod.rs:6729]: I have no idea why, but adding this branch causes SO in the recursion tests such as `parse_deeply_nested_parens_hits_recursion_limits`

Comment (universalmind303) [reply src/parser/mod.rs:6729]: even if the branch is never met, the presence of it is causing SO.

Comment (universalmind303) [reply src/parser/mod.rs:6729]: maybe @alamb, @andygrove @nickolay knows why adding this condition causes a SO.

Comment (alamb) [reply src/parser/mod.rs:6729]: This branch probably makes the stack frame size increase in debug builds as it adds some new local variables. I think we could fix the test by decreasing the recursion limit

File: tests/sqlparser_duckdb.rs [hunk 2]
```diff
@@ -244,3 +246,188 @@ fn test_duckdb_load_extension() {
...
+        qualify: None,
+        from_before_select: true,
+    };
+    assert_eq!(stmt, expected);
+}
+
+#[test]
+fn test_duckdb_from_statement_with_filter_and_select() {
+    let stmt = duckdb().verified_only_select("FROM t1 SELECT b WHERE a = 1");
+    println!("{:?}", stmt);
```
Comment (alamb) [inline tests/sqlparser_duckdb.rs:333]: left over?

File: tests/sqlparser_duckdb.rs [hunk 3]
```diff
@@ -244,3 +246,188 @@ fn test_duckdb_load_extension() {
...
+        qualify: None,
+        from_before_select: true,
+    };
+    assert_eq!(stmt, expected);
+}
+
+#[test]
+fn test_duckdb_from_statement_with_filter() {
+    let stmt = duckdb().verified_only_select("FROM t1 WHERE a = 1");
+    println!("{:?}", stmt);
```
Comment (alamb) [inline tests/sqlparser_duckdb.rs:289]: left over?

File: src/parser/mod.rs [hunk 4]
```diff
@@ -6936,6 +6942,162 @@ impl<'a> Parser<'a> {
             having,
             named_window: named_windows,
             qualify,
+            from_before_select: false,
+        })
+    }
+
+    /// Parse a duckdb style `FROM` statement without a select.
+    /// assuming the initial `FROM` was already consumed
+    pub fn parse_from_select(&mut self) -> Result<Select, ParserError> {
```
Comment (alamb) [inline src/parser/mod.rs:6951]: this seems largely copied from the normal select parsing -- is there any way to refactor the code to reduce the duplication

**Review by alamb — COMMENTED** [review]
Thank you for this contribution @universalmind303  -- I left some comments that hopefully will help us get this PR good to go

**Comment by alamb** [comment]
Marking as Draft to signify this PR is no longer waiting on review. Please mark it as ready for review when it is ready for another look.

**Comment by alamb** [comment]
FWIW @Nikita-str  has solved the stack overflow issues in https://github.com/sqlparser-rs/sqlparser-rs/pull/1171 if you wanted to revive this PR.

**Pull request closed without merging**